packages/core-bridge/ts/worker-tuner.ts

```diff
@@ -69,3 +87,102 @@ export interface FixedSizeSlotSupplier {
   // The maximum number of slots that can be issued
   numSlots: number;
 }
+
+/**
+ * The interface can be implemented to provide custom slot supplier behavior.
+ */
+export interface CustomSlotSupplier<SI extends SlotInfo> {
```

Comment:
I noticed that a "maximum slots" getter isn't here. I think that's a good thing, just noticing the mismatch with Go/Java.